lib/connection/commands.js

```diff
@@ -419,9 +421,46 @@ var Response = function(bson, data, opts) {
   this.responseTo = data[this.index] | data[this.index + 1] << 8 | data[this.index + 2] << 16 | data[this.index + 3] << 24;
   this.index = this.index + 4;
 
-  // Skip op-code field
+  // Read op-code field
+  this.opCode = data[this.index] | data[this.index + 1] << 8 | data[this.index + 2] << 16 | data[this.index + 3] << 24;
   this.index = this.index + 4;
 
+  // If the message we get is OP_COMPRESSED, then we have to uncompress the data 
+  if (this.opCode == 2012) {
```

Comment:
nit: I would prefer if we didn't use this magic number here (opcode 2012), rather name it as a constant above